sdk/search/azure-search/src/main/java/com/azure/search/SearchApiKeyPipelinePolicy.java

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Mono;
+
+import java.util.Objects;
+
+/**
+ * Policy that adds the Cognitive Search Service api-key into the request's Authorization header.
+ */
+public class SearchApiKeyPipelinePolicy implements HttpPipelinePolicy {
+
+    private final SearchApiKeyCredential apiKey;
+
+    /**
+     * Constructor
+     *
+     * @param apiKey Azure Cognitive Search service api admin or query key
+     * @throws IllegalArgumentException when the api key is an empty string
+     */
+    public SearchApiKeyPipelinePolicy(SearchApiKeyCredential apiKey) {
+        Objects.requireNonNull(apiKey);
+        this.apiKey = apiKey;
+    }
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        context.getHttpRequest()
+            .setHeader("api-key", this.apiKey.getApiKey());
```

Comment:
Is api-key a secret by any chance? Should this value be redacted in logging and playback recording?

Reply:
Currently api-key is not a secert and is exposed in the azure portal 
The api-key value is not exposed in the test session recordings.

Reply:
Awesome! Just wanted to verify that we wouldn't be leaking any secrets if this happened to get logged.